### DISCUSSION TITLE
FEVM vs Celestia or Smartweave

### DISCUSSION THREAD

**Discussion by hdahme** [filecoin-project/community#608] [Developers]
Hi! Sorry I'm late to the party, wanted to do my homework before asking such a question, but -- I'm having trouble wrapping my mind around how FEVM scales and doesn't become another Polkadot (ie, limited execution slots => $$ parachain auctions). The only other tenable solution I've seen is to use lazy execution to verify contracts, and treat IPFS/Arweave/Celestia's as a permanent  data availability layer for light nodes. So my question is..
1. Does FVM use lazy execution - and then how's the compatibility with other EVM/SolVM paradigms?
2. If not, how does FVM change the hardware requirements for FIL node operators, and what're the implications of that? ie, do FVM node operators all of a sudden need to use machines with a ton of space _and_ CPUs? Or FVM really a marriage between [bacalhau.org](https://www.bacalhau.org/) and IPFS?
3. Please can you link me to `filecoin.sol` referenced in [this](https://www.youtube.com/watch?v=EqLlyJ13DLY) video? Would love to look at how to access storage primitives from inside contracts themselves

Otherwise, this is huge, congrats on the launch!! 👏

**Comment by hdahme** [comment]
Asked in Slack, please disregard

**Comment by jnthnvctr** [comment]
1. FVM does not use lazy execution. 
2. The FVM doesn't change hw requirements for FIL nodes - but it also isn't where you'd run things like Bacalhau! This blog might be helpful: https://medium.com/tldrfilecoin/state-and-direction-of-filecoin-summarized-4b90c59e3cca . Longer term -  you should look at Interplanetary Consensus (some great youtube talks on this too!). https://fil.space is where they're tracking work.
